README.adoc

```diff
@@ -0,0 +1,154 @@
+Besides adding the text, order of the lines can be reversed.
+
+== Changes in the addon
+
+This version of the addon works with Anaconda 32.19 and brings an important change
+where the code no longer runs in a single process. Instead, there are now two parts:
+The GUI or TUI, and worker service. These two processes communicate using D-Bus.
+This method is used in Anaconda itself.
+
+== Additional information
+
+For information on developing addons see: https://rhinstaller.github.io/anaconda-addon-development-guide/
+
+
+= File and directory overview
+
+The following diagram explains the structure of files and directories needed for an addon. All items of importance are explained below the diagram. The layout actually mostly follows structure of Anaconda's code.
+
+----
+<repository root>
+├── dbus-data <1>
+├── LICENSE
+├── Makefile
+├── README.adoc
+└── org_fedora_hello_world <2>
+    ├── constants.py <3>
+    ├── dbus_service <4>
+    ├── categories <5>
+    ├── gui <6>
+    └── tui <7>
+----
+<1> Files needed to configure and run the service
+<2> Directory for actual Python code of the addon
+<3> Constants shared by the service and GUI code
+<4> Code that runs as a stand-alone service/process
+<5> Code that describes how to integrate the GUI or TUI into Anaconda's interface
+<6> Code for graphical interface
+<7> Code for text interface
+
+In most cases, you can assume that `HelloWorld`, `hello_world` and other variations of "__hello world__" are specific to this addon and will need changing for your own addon.
+
+== Service files
+
+The files in the `dbus-data` directory are needed to configure and run the service in the installation environment. These are:
+
+`org.fedoraproject.Anaconda.Addons.HelloWorld.conf`::
+Configuration for D-Bus that describes the objects on the bus and access rights to these.
+
+`org.fedoraproject.Anaconda.Addons.HelloWorld.service`::
+A `systemd` unit file that describes the service to be started. Note that the file is not actually interpreted by `systemd`, so some syntax features will not work! Instead, a process that sets up the D-Bus reads this.
+
+`org.fedoraproject.Anaconda.Addons.HelloWorld.sh`::
+A shell script that sets up environment for `python3` and runs the code in `dbus_service/__main__.py`.
+
+The name of the directory is not required by any rule, but make sure to adjust the `Makefile` if you change the name.
+
+== Makefile
+
+The `Makefile` provided with this addon is very basic. It copies files to their respective paths, and then creates an updates image that contains these files.
+
+The paths for the various types of files encoded in the `Makefile` are required by Anaconda. If you put your own files anywhere else, the addon will not work.
+
+Note that the updates image uses the same format as `initrd` images, but is handled by Anaconda, so caveats apply. In particular, addon images cannot be concatenated like `initrd` ones - you will have to unpack these, merge the trees, and re-compress.
```

Comment:
It might be good to point to the de-facto updates image docs there: https://fedoraproject.org/wiki/Anaconda/Updates